kubeadm/v1.15.0/KubeCluster.ps1

```diff
@@ -0,0 +1,273 @@
+PrintConfig
+WriteKubeClusterConfig
+
+
+# Initialize internal network modes of windows corresponding to 
+# the plugin used in the cluster
+$Global:NetworkName = "cbr0"
+$Global:NetworkMode = "l2bridge"
+if ($Global:NetworkPlugin -eq "vxlan")
+{
+    $Global:NetworkMode = "overlay"
+    $Global:NetworkName = "vxlan0"
+}
+######################################################################################################################
+
+# Handle --install
+if ($install.IsPresent)
+{
+    InstallContainersRole
+    if (!(Test-Path $env:HOMEDRIVE/$env:HOMEPATH/.ssh/id_rsa.pub))
+    {
+        $res = Read-Host "Do you wish to generate a SSH Key & Add it to the Linux control-plane node [Y/n] - Default [Y] : "
+        if ($res -eq '' -or $res -eq 'Y'  -or $res -eq 'y')
+        {
+            ssh-keygen.exe
+        }
+    }
+
+    $pubKey = Get-Content $env:HOMEDRIVE/$env:HOMEPATH/.ssh/id_rsa.pub
+    Write-Host "Execute the below commands on the Linux control-plane node ($Global:MasterIp) to add this Windows node's public key to its authorized keys"
+    
+    Write-Host "touch ~/.ssh/authorized_keys"
+    Write-Host "echo $pubKey >> ~/.ssh/authorized_keys"
+
+    $res = Read-Host "Continue to Reboot the host [Y/n] - Default [Y] : "
+    if ($res -eq '' -or $res -eq 'Y'  -or $res -eq 'y')
+    {
+        Restart-Computer -Force
+    }
+
+    InstallCRI $Global:Cri
+    InstallKubernetesBinaries -Destination  $Global:BaseDir -Source $Global:ClusterConfiguration.Kubernetes.Source
+
+    exit
+}
+
+# Handle -Join
+if ($Join.IsPresent)
+{
+    $kubeConfig = GetKubeConfig
+    if (!(KubeConfigExists))
+    {
+        # Fetch KubeConfig from the master
+        DownloadKubeConfig -Master $Global:MasterIp -User $Global:MasterUsername
+        if (!(KubeConfigExists))
+        {
+            throw $kubeConfig + " does not exist. Cannot connect to the control-plane node"
+        }
+    }
+
+    # Validate connectivity with the API Server
+
+    Write-Host "Trying to connect to the Kubernetes control-plane node"
+    try {
+        ReadKubeClusterInfo 
+    } catch {
+        throw "Unable to connect to the control-plane node. Reason [$_]"
+    }
+
+    $KubeDnsServiceIP = GetKubeDnsServiceIp
+    $ClusterCIDR = GetClusterCidr
+    $ServiceCIDR = GetServiceCidr
+    
+    Write-Host "############################################"
+    Write-Host "Able to connect to the control-plane node"
+    Write-Host "Discovered the following"
+    Write-Host "Cluster CIDR    : $ClusterCIDR"
+    Write-Host "Service CIDR    : $ServiceCIDR"
+    Write-Host "DNS ServiceIp   : $KubeDnsServiceIP"
+    Write-Host "############################################"
+
+    #
+    # Install Services & Start in the below order
+    # 1. Install & Start Kubelet
+    InstallKubelet  -CniDir $(GetCniPath) `
+                -CniConf $(GetCniConfigPath) -KubeDnsServiceIp $KubeDnsServiceIp `
+                -NodeIp $Global:ManagementIp -KubeletFeatureGates $KubeletFeatureGates
```

Comment:
Where's InstallKubelet implemented? It needs to set resolv-conf="" on Windows, otherwise you get an error starting the pods such as:

```

Events:
  Type     Reason                  Age               From               Message
  ----     ------                  ----              ----               -------
  Normal   Scheduled               24s               default-scheduler  Successfully assigned default/iis-2019-5cd8f558dc-xjxhl to win1
  Warning  FailedCreatePodSandBox  9s (x2 over 23s)  kubelet, win1      Failed create pod sandbox: open /etc/resolv.conf: The system cannot find the file specified.
```

Reply:
I think the --resolv-conf parameter is set in the config file now, right?

Reply:
Added it to the new-service kubelet call